### PR TITLE
feat(mcp): add docvet mcp CLI subcommand and integration tests

### LINKED ISSUE #266: Story 29.2: CLI Wiring and Tests
## Story

As a **developer setting up docvet's MCP server**,
I want a `docvet mcp` subcommand that starts the MCP server,
so that I can configure MCP clients to connect to docvet easily.

## Acceptance Criteria

1. `docvet mcp` starts MCP server on stdio when `mcp` dependency installed
2. `docvet mcp` without `mcp` package shows clear error message and exits with code 1
3. MCP `docvet_check` response matches `docvet check --format json` structure
4. MCP `docvet_rules` lists all 20 rules with name, check, description, category
5. Server shuts down cleanly on client disconnect

---

Story file: `_bmad-output/implementation-artifacts/29-2-cli-wiring-and-tests.md`
Branch: `feat/mcp-29-2-cli-wiring-and-tests`

### PR DESCRIPTION
**Pull request by Alberto-Codes**
The MCP server (Story 29.1) had no CLI entry point. Users needed a `docvet mcp` subcommand to start the server, matching the existing `docvet lsp` pattern.

- Add `mcp()` subcommand to `cli.py` with lazy import and dependency error handling
- Add 4 CLI unit tests (help, missing dep, invocation, structural registration)
- Add 6 MCP integration tests using the `mcp` client SDK (e2e check, rules, parity, shutdown, zero findings, explicit checks)
- Add `docvet lsp` and `docvet mcp` entries to CLI reference docs (backfills missing `lsp` docs from Epic 24)

Test: `uv run pytest tests/unit/test_cli.py::TestMcpSubcommand tests/integration/test_mcp.py -v`

Closes #266

fix(mcp): address code review findings for story 29.2

test(mcp): add by_check consistency assertion to parity test

---

### Checklist
- [x] Self-reviewed my code
- [x] Tests pass (`uv run pytest`)
- [x] Lint passes (`uv run ruff check .`)
- [x] Types pass (`uv run ty check`)
- [x] Breaking changes use `!` in title and `BREAKING CHANGE:` in body

### Review Focus
- `cli.py:1268-1287`: `mcp()` follows `lsp()` pattern exactly — lazy import with `start_server as mcp_start_server` alias to avoid shadowing the function name
- `tests/integration/test_mcp.py`: Uses `mcp` client SDK (`stdio_client` + `ClientSession`) for true end-to-end testing. Parity test validates MCP findings match CLI JSON output.

### Related
- feat(mcp): add core MCP server with docvet_check and docvet_rules tools #261

### CODE REVIEW

File: tests/integration/test_mcp.py [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+                    await asyncio.wait_for(
+                        session.initialize(), timeout=_SERVER_TIMEOUT
+                    )
+                    # Call a tool to confirm server is working
+                    await _call_tool(session, "docvet_rules")
+                    # Session and client exit cleanly via context managers
+            # If we get here without error, the server shut down cleanly
+            return True
+
+        assert asyncio.run(_run()) is True
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply tests/integration/test_mcp.py]: Addressed in latest push — wrapped `_run()` in `asyncio.wait_for(..., timeout=2 * _SERVER_TIMEOUT)` to cover context-manager exit. Used 2x timeout (60s) instead of 1x to avoid false kills when `initialize()` + `call_tool()` each consume part of the single timeout budget. Applied to shutdown test only since it's the test whose purpose is verifying teardown behavior.
